### PR TITLE
Let `testValidate` infer aten outputs

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: csrc/executor_utils.cpp [hunk 1]
```diff
@@ -832,6 +832,10 @@ void bindInputForExprEvaluation(
     ExpressionEvaluator& expr_eval) {
   if (val->getValType() == ValType::TensorView) {
     TensorView* cg_tensor = val->as<TensorView>();
+    auto tensor_arg_abstract = dynamic_cast<const TensorArgAbstract*>(arg);
+    if (tensor_arg_abstract != nullptr) {
+      expr_eval.bind(cg_tensor, tensor_arg_abstract->getTensor());
```
Comment (naoyam) [inline csrc/executor_utils.cpp:837]: Here, we are binding the metadata of `tensor_arg_abstract`?

Comment (zasdfgbnm) [reply csrc/executor_utils.cpp:837]: We are binding `tensor_arg_abstract->getTensor()`, which is an `at::Tensor`.

Comment (naoyam) [reply csrc/executor_utils.cpp:837]: I guess what I'm wondering is when an aten tensor is bound to a TensorView, are we binding its metadata as well? If so, can we remove the code below this part?

File: test/validator.h [hunk 2]
```diff
@@ -444,5 +452,27 @@ void testValidate(
   }
 }
 
+// The variant with automatically inferred aten outputs
```
Comment (naoyam) [inline test/validator.h:455]: Please mention that the given fusion is assumed to contain only supported expressions for ExpressionEvaluator.

File: csrc/executor_utils.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/executor_utils.cpp:837]: Yes, when binding an aten tensor, we should have all its metadata. The code below should be removed, but needs more cleanup (which will be done in my future PR) in order to do so. For example, tensor sizes are currently still named scalar `T1.size[0]`.

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**